### PR TITLE
Add `README.md` files to important directories

### PR DESCRIPTION
**Pull request by namurphy**
If a directory has a `README.md` file, it shows up as documentation when you navigate to that directory on GitHub. I decided that it would be worthwhile to add `README.md` files to our most important directories ahead of the summer school. This way, new contributors would be able to quickly get some idea of what the contents of the directory are for. 

My _hope_ is to keep these succinct and refer people to the appropriate pages in the contributor guide so that we don't need to spend much time maintaining this. I'm definitely open to ways to shorten the content.

The files I added were:
 - `src/plasmapy/README.md`
 - `tests/README.md`
 - `docs/README.md`
 - `type_stubs/README.md`
 
In addition, I updated the primary `README.md` with some minor fixes (fixing broken badges, removing redundant text, and dropping mention of Anaconda Navigator because it has new license restrictions).

I also added `requirements.txt` to `.gitignore` (as requested in #2647) so that contributors can use that file to define their own environments and that this file doesn't accidentally get committed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,13 +212,6 @@ bring up discussion topics, and ask questions.
 You can subscribe to PlasmaPy's low-volume [mailing list] to receive
 PlasmaPy newsletters and other announcements.
 
-### Suggestion box
-
-We have a [suggestion box] if you would like to (optionally
-anonymously) suggest a feature/topic for consideration. These
-suggestions might be changed into GitHub issues for further
-discussion.
-
```
Comment (namurphy) [inline README.md:222]: Removing this since it's been pretty rare for us to get suggestions in the suggestion box.

File: .gitignore [hunk 2]
```diff
@@ -132,6 +132,10 @@ auto_examples
 docs/about/_authors.rst
 docs/notebooks/*.md
 
+# Allow contributors to create a custom requirements.txt file for
+# their own use.
+requirements.txt
+
```
Comment (namurphy) [inline .gitignore:138]: As requested in #2647.

File: README.md [hunk 3]
```diff
@@ -82,32 +80,30 @@ recent presentations are available from the [PlasmaPy Community on
...
-> likely need to be redone by running `pip install -e .[tests,docs]` in
-> the top-level directory of the repository.
+> top-level [`tests/`] directory. Existing pull requests should pull
+> in the changes from the `main` branch with `git pull upstream main`
+> (assuming the remote for the primary PlasmaPy repo is named
+> `upstream`). Because `git` does not automatically remove
+> directories, the `plasmapy/` directory in older clones must be
+> manually deleted. If you previously did an editable installation of
+> PlasmaPy, it will likely need to be redone by running `pip install
+> -e .[tests,docs]` in the top-level directory of the repository.
```
Comment (namurphy) [inline README.md:106]: 🎨 I only justified the text here. No substantive changes.

File: README.md [hunk 4]
```diff
@@ -130,21 +126,14 @@ py -3.12 -m pip install plasmapy
...
 
-If you have [installed Conda], then you can also install PlasmaPy into
-an activated Conda environment by running:
+If you have [installed Conda], then you can install PlasmaPy into  an
+activated Conda environment by running:
 ```Shell
 conda install -c conda-forge plasmapy
 ```
-PlasmaPy can also be installed using [Anaconda Navigator] so long as
-`conda-forge` is added as a channel.
```
Comment (namurphy) [inline README.md:139]: The licensing for Anaconda Navigator has gradually been becoming more restrictive, so I'm leaning against including it in `README.md` (though it should be discussed in the online documentation).

File: type_stubs/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Type stubs
+
+[mypy]: https://mypy.readthedocs.io
+[type hint annotations]: https://docs.python.org/3/glossary.html#term-type-hint
+[type stub files]: https://docs.python.org/3/library/typing.html#stub-files
```
Comment (JaydenR2305) [inline type_stubs/README.md:5]: The "type stub files" link isn't working for me. Maybe https://mypy.readthedocs.io/en/stable/stubs.html instead?

File: README.md [hunk 6]
```diff
@@ -130,21 +126,14 @@ py -3.12 -m pip install plasmapy
 The `3.12` may be replaced by any version of Python that is supported by
 PlasmaPy.
 
-If you have [installed Conda], then you can also install PlasmaPy into
-an activated Conda environment by running:
```
Comment (namurphy) [inline README.md:134]: 😅 One of the steps I take when editing a document is to remove as many occurrences of "also" as I can!

File: type_stubs/README.md (hunk 5 above)
Comment (namurphy) [reply type_stubs/README.md]: Fixed!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (namurphy) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I've been noticing that most PRs that get submitted end up deleting the suggested sections of the pull request template, including most of the PRs that I submit!  This PR reduces the amount of information to the essentials, which incidentally makes it easier to delete!

(A bit of scope creep for this PR, but it's still related to the main idea.)

Comment (namurphy) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I've noticed that when most contributors make pull requests (including me!), the pull request template gets deleted. Drastically shortening the template will make it more likely that the contents actually do get read (in particular the link to the contributor guide)...and it'll also make it easier to delete!

(This was a bit of scope creep for this PR, but it still fits in with the general idea.)
